src/lib/ng-package/entry-point/write-package.transform.ts

```diff
@@ -132,7 +132,7 @@ async function writePackageJson(
   try {
     checkNonPeerDependencies(packageJson, 'dependencies', whitelist);
   } catch (e) {
-    await rimraf(entryPoint.destinationPath);
+    await rimraf(entryPoint.libraryDestinationPath);
```

Comment:
Interesting, I think that I've introduced a regression that the tests didn't catch, I'll look into this and write a test if that's the case.

Reply:
Here, the old code caught the error, removed entry point destination directory and then the error was re-thrown in order to cancel the build.

I think that this change is better, because the whole library destination directory is removed and you're not left with a partial build.

Reply:
I think deleting the entire library makes sense in this case.

Reply:
Would be great if you can do the rename as a separate commit.

Reply:
Moved to a separate PR: #1845